### PR TITLE
feat: enable hierarchical namespace

### PR DESCRIPTION
**Pull request by stiross79**

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -43,6 +43,24 @@ variable "shared_access_key_enabled" {
...
+  description = "The encryption type of the queue service. Possible values are Service and Account"
+  type        = string
+  default     = "Service"
+}
+
+variable "table_encryption_key_type" {
+  description = "The encryption type of the table service. Possible values are Service and Account"
+  type        = string
+  default     = "Service"
+}
```
Comment (helenakallekleiv) [inline variables.tf:62]: We should consider making these variables optional with default argument being `null` or similar. When running Go test on all four examples, the argument for the variables is set to `Service`. In a scenario e.g. running Terraform Import on existing resources that does not include these values, an option to exclude them would be ideal.

Comment (helenakallekleiv) [reply variables.tf:62]: @hknutsen Tagging you here so are in the loop, and perhaps could give your opinion on how we could implement this 😄

Comment (hknutsen) [reply variables.tf]: These values are always set. According to docs, they are set to `"Service"` by default. Makes sense to me to set them to the same default value in our module.

**Comment by hknutsen** [comment]
@helenakallekleiv I added relevant examples and tests. Could you have a look when you have the time? 🙂

File: variables.tf (hunk 1 above)
Comment (helenakallekleiv) [reply variables.tf]: So even for standard storage accounts, the values are always by default set to "Service", even if the storage account is not ment as a data lake account?

Comment (hknutsen) [reply variables.tf]: Yes, they don't seem to have anything to do with Data Lake Storage actually, but I'm guessing Stian and co. are setting different values in some of their Data Lake Storage accounts, which is why it's included in this PR 🙂 

All tests passed as well 🎉

**Comment by hknutsen** [comment]
All tests passed 🎉

**Review by helenakallekleiv — APPROVED** [review]
All tests passed for me as well 🥳

**Pull request merged**